app/client/src/store/actions.ts

```diff
@@ -45,17 +45,33 @@ export default {
   },
   async runPoppunk(context: ActionContext<RootState, RootState>) {
     const { state, commit } = context;
-    const jsonSketches = {} as { [key: string]: Record<string, never> };
-    // join all file hashes to create a project hash
-    const phash = Md5.hashStr(Object.keys(state.results.perIsolate).sort().join());
+    // generate filenameMapping that the backend can use to replace
+    // filehashes with filenames in results.
+    // Also generate a string of ordered hashes and corresponding filenames
+    // to be used to generate a projecthash that is unique to this combination
+    // of file contents and filenames
+    const filenameMapping = {} as { [key: string]: string | undefined };
+    let mappingOrdered = '';
+    Object.keys(state.results.perIsolate).sort().forEach((filehash) => {
+      filenameMapping[filehash] = state.results.perIsolate[filehash].filename;
+      mappingOrdered += filehash;
+      mappingOrdered += state.results.perIsolate[filehash].filename;
+    });
+    const phash = Md5.hashStr(mappingOrdered);
     commit('setProjectHash', phash);
+    // add all sketches to object
+    const jsonSketches = {} as { [key: string]: Record<string, never> };
```

Comment:
Why `never`? Is there a not a real type for these? I'm not really sure why this line has changed, is it related to the file name mapping?